### DISCUSSION TITLE
Rtables rendering in R shiny

### DISCUSSION THREAD

**Discussion by BhargavAchanta** [insightsengineering/rtables#833] [General]
Hi, How can i render a table created using rtables using Render function in R shiny. I have tried renderTable, but it doesnt seem to work. 
any work around to convert Rtables in to data frame?

Thanks

**Comment by shajoezhu** [comment]
hi @BhargavAchanta , check out the function `rtables::as_result_df()` i think that is what you want

**Reply by BhargavAchanta** [reply]
Hello,
I am trying to convert this table (fig.1) created using rtables. I have used the function you have provided but it does not work
<img width="706" alt="df1" src="https://github.com/insightsengineering/rtables/assets/158379030/8e55d2ef-a982-4dc4-add1-21c89e704916">
 
Here is my code in the server function. 

aeptbodysys <- build_table(lyt, df = adae, alt_counts_df = adsl)
aeptbodysys1 <- rtables::as_result_df(aeptbodysys)
output$table <- renderDataTable(aeptbodysys1)
 
in the UI function I am using "table" object as follows and I see this (fig.2) in the shiny app which is not what i wanted.

ui <- page_sidebar(
  title = "My dashboard",
  sidebar = sidebar(selectInput("length","length",choices = iris$Sepal.Length)
),
dataTableOutput('table')  # "table" object created in server function.
)

fig.2
<img width="487" alt="df" src="https://github.com/insightsengineering/rtables/assets/158379030/e35ce0e2-c23c-45a2-b3d3-720f68b86fb7">

**Reply by Melkiades** [reply]
use `as_is = TRUE`. Otherwise, there is information there that keeps track of the structure of the table

**Comment by BhargavAchanta** [comment]
where does this as_is = TRUE  go? , i have tried, but gives me an error

**Comment by shajoezhu** [comment]
hi @BhargavAchanta , can you install rtables from the main branch and test. thanks
